CMakeLists.txt

```diff
@@ -55,3 +54,5 @@ add_subdirectory(cpp-lib)
 # compiler itself
 add_subdirectory(compiler)
 
+target_compile_options(compiler PRIVATE -Wno-deprecated-declarations)
```

Comment:
Whats the purpose of disabling warnings? The code compiles error free using clang (llvm) and the vs compilers?
If this is needed for other compilers we should be specific on which but i would not want to disable deprecation warnings for all compilers